File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int idx = 0, cnt = 1, i = 0;
        int n = chars.size();

        for (i = 1; i < n; i++) {
            if (chars[i] == chars[i - 1]) {
                cnt++;
            } else {
                chars[idx++] = chars[i - 1];
                if (cnt > 1) {
                    for (char c : to_string(cnt)) {
                        chars[idx++] = c;
                    }
                }
                cnt = 1;
            }
        }

        chars[idx++] = chars[i - 1];  // Add the last character
        if (cnt > 1) {
            for (char c : to_string(cnt)) {
                chars[idx++] = c;
            }
        }

        return idx;
    }
};

File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        idx, cnt = 0, 1
        i = 0
        for i in range(1, len(chars)):
            if (chars[i] == chars[i - 1]):
                cnt += 1

            elif (cnt == 1):
                chars[idx] = chars[i - 1]
                idx += 1

            elif (cnt != 1):
                chars[idx] = chars[i - 1]
                idx += 1
                for c in str(cnt):
                    chars[idx] = c
                    idx += 1
                cnt = 1
        
        if (cnt == 1):
            chars[idx] = chars[i]
            idx += 1
        
        elif (cnt != 1):
            chars[idx] = chars[i-1]
            idx += 1
            for c in str(cnt):
                chars[idx] = c
                idx += 1

        return idx